.github/workflows/java-model-diff.yml

```diff
@@ -0,0 +1,98 @@
+name: Models as Data
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      projects:
+        description: "The projects to generate models for"
+        required: true
+        default: '["netty/netty"]'
+  pull_request:
```

Comment:
How strict do you want to keep this? Actions runs from external contributors will not start unless we manually approve them, but it's good to be careful. Should this run only for maintainers, or only from PRs based within this repository (no forks)?

Reply:
We restrict it to only this repo now. Generally, I think external contribtors might find this useful but it's too early to invest into that. Is there anything else we need to only run this for maintainers for now?

Reply:
`if: github.repository == 'github/codeql'` is true even for PRs opened against this repo from forks. Let's chat separately so I can understand exactly which PRs you want this to run on.

Reply:
@adityasharad Could you clarify you security concerns? The `pull_request` trigger is not particularly dangerous because pull requests from forks only run with (short-lived) read-only tokens.